cmd/kubeadm/app/cmd/phases/saveconfig.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	kubeadmapiext "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha1"
+	"k8s.io/kubernetes/cmd/kubeadm/app/phases/saveconfig"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	configutil "k8s.io/kubernetes/cmd/kubeadm/app/util/config"
+	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
+)
+
+// NewCmdSaveConfig returns the Cobra command for running the saveconfig phase
+func NewCmdSaveConfig() *cobra.Command {
+	var cfgPath, kubeConfigFile string
+	cmd := &cobra.Command{
+		Use:     "save-config <node-name>",
+		Short:   "Upload the currently used configuration for kubeadm to a ConfigMap in the cluster.",
+		Aliases: []string{"saveconfig"},
+		Run: func(_ *cobra.Command, args []string) {
```

Comment:
I don't know if we have this pattern for the other commands, but if we're not expecting anything in `args`, I would error in the case anything is provided. Someone might run `kubeadm phase saveconfig my-file` thinking that they're overriding the filename to use, but not realize that they forgot `--config` and accidentally use the wrong file.